tls/s2n_server_hello.c

```diff
@@ -45,7 +45,7 @@ int s2n_server_hello_recv(struct s2n_connection *conn)
 
     GUARD(s2n_stuffer_read_bytes(in, protocol_version, S2N_TLS_PROTOCOL_VERSION_LEN));
 
-    conn->server_protocol_version = (protocol_version[0] * 10) + protocol_version[1];
+    conn->server_protocol_version = (uint8_t)(protocol_version[0] * 10) + protocol_version[1];
```

Comment:
I'm surprised we need these casts

Reply:
I’m constantly surprised about c compilers catching pedantic cast issues and missing important ones.

But this popped as a warning on my al2012 box.